pkg/provider/resource_cluster.go

```diff
@@ -87,6 +87,13 @@ func (c *ClusterResource) Schema() *schema.Resource {
 					},
 				},
 			},
+
+			"cluster_type": {
```

Comment:
We need to write [a state migration](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/state-migration#terraform-v0-12-sdk-state-migrations) for adding a new field to the biganimal_cluster schema. 

You can test the state migration as follows:
* Create a biganimal_cluster using the latest terraform-provider-biganimal.
* Make sure you have the cluster information in the terraform state. You can run `terraform plan/apply` after a successful cluster creation, and it says there are no changes to apply. 
* After that, you compile your local code and start using the compiled terraform-provider-biganimal that includes this `cluster_type` change and try running `terraform plan` on the existing biganimal_cluster resource. 
* Because it will read the terraform state first, it won't find the `cluster_type` field in the `biganimal_cluster` schema. The state migration is necessary at that step.

Reply:
Btw, I tested this scenario and we don't need the state migration, because we're setting the cluster_type during the read function.  

I'm resolving this conversation.